shared/route-tree/render-route.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+
+function renderRouteNode(RouteComponent, routeTree, path, setRouteState, leafTags, child) {
+  return <RouteComponent
+    routeProps={routeTree.staticProps.merge(routeTree.props).toJS()}
```

Comment:
Good point! This is hacky. I seem to recall you could pass immutable objects directly to react components. I'll look into this.